src/scripts/Game.ts

```diff
@@ -191,10 +191,15 @@ class Game {
         }
 
         this.interval = setInterval(() => !this.worker || !Settings.getSetting('useWebWorkerForGameTicks').value ? this.gameTick() : null, GameConstants.TICK_TIME);
+        window.onbeforeunload = () => {
+            player._lastSeen = Date.now();
+            Save.store(player);
```

Comment:
maybe externalize the saving of the player to a private function, since this code would now be duplicated

Reply:
done